packages/trip-form/src/GeneralSettingsPanel/index.js

```diff
@@ -97,6 +106,38 @@ GeneralSettingsPanel.propTypes = {
   // method getQueryParamProperty from "@opentripplanner/core-utils/query".
   // eslint-disable-next-line react/forbid-prop-types
   query: PropTypes.any,
+  /**
+   * An object {parameterName: text, "parameterName.options": [array]}
+   * whose attributes correspond to query parameters. This allows to customize the
+   * labels and options (both text and values).
+   * For query parameter names and value formats,
+   * see https://github.com/opentripplanner/otp-ui/blob/master/packages/core-utils/src/__tests__/query.js#L14
+   */
+  queryParamMessages: PropTypes.shape({
+    from: PropTypes.string,
+    to: PropTypes.string,
+    maxWalkDistance: PropTypes.string,
+    "maxWalkDistance.options": queryParamOptionsType,
+    maxBikeDistance: PropTypes.string,
+    "maxBikeDistance.options": queryParamOptionsType,
+    optimize: PropTypes.string,
+    "optimize.options": queryParamOptionsType,
+    optimizeBike: PropTypes.string,
+    "optimizeBike.options": queryParamOptionsType,
+    maxWalkTime: PropTypes.string,
+    "maxWalkTime.options": queryParamOptionsType,
+    walkSpeed: PropTypes.string,
+    "walkSpeed.options": queryParamOptionsType,
+    maxBikeTime: PropTypes.string,
+    "maxBikeTime.options": queryParamOptionsType,
+    bikeSpeed: PropTypes.string,
+    "bikeSpeed.options": queryParamOptionsType,
+    maxEScooterDistance: PropTypes.string,
+    "maxEScooterDistance.options": queryParamOptionsType,
+    watts: PropTypes.string,
+    "watts.options": queryParamOptionsType,
+    wheelchair: PropTypes.string
+  }),
```

Comment:
I'm not a huge fan of this since it pushes the responsibility of providing messages to the implementing application instead of maintaining messages within this package.